Port GetMailboxesSimulation to RFC-8621

Arsnael opened the pull request.

Arsnael committed:
ISSUE-84 Rename jmap packages to jmap.draft

Arsnael committed:
ISSUE-84 MailboxGet Scenario & Simulation

Arsnael committed:
ISSUE-84 Use linagora/james-memory:branch-master for IT tests

Arsnael committed:
ISSUE-84 Implement SessionStep

Arsnael committed:
ISSUE-84 Retrieve AccountId from SessionStep

Arsnael committed:
ISSUE-84 Mailbox/get step scenario & simulations

chibenwa commented:
Update the readme accordingly (adding `.draft` in the FQDN, mention the new simulation if needed)

Arsnael committed:
ISSUE-84 Update README with new JMAP-RFC8621 simulations section

chibenwa commented on README.adoc:
```diff
@@ -108,7 +110,22 @@ Authenticate 100 JMAP users, one time. Then they all retrieve their system mailb
+
+.scala
+----
+org.apache.james.gatling.simulation.jmap.draft.JmapMessageUpdateSimulation
```
Wrong FQDN for being in MailboxGetSimulation section I bet

Arsnael replied:
Too quick of a bad copy/paste... Thanks!

Arsnael commented:
```
Request 'getMailQueues:spool' failed for user 1: status.find.is(200), but actually found 404
[...]
[error] Failed tests:
[error]         org.apache.james.gatling.jmap.draft.JmapQueueBrowseIT
```
From what I could see when debugging, it's because we change the image of James to `branch-master`. it seems before when starting James we had some queues created, like the one we are trying to check here 'spool', but not anymore. What changed for memory?

Arsnael committed:
ISSUE-84 Disable JmapQueueBrowseIT integration test

404 on trying to fetch the mail queue 'spool' with the upgraded James memory image to branch-master.
It seems there is potentially a bug where we loose the mail queues on memory version lately.
Need to fix it on James before re-enabling this test.

chibenwa commented on src/it/scala-2.12/org/apache/james/gatling/jmap/draft/JmapQueueBrowseIT.scala:
```diff
@@ -1,20 +1,26 @@
+/*
+ISSUE-84: 404 on trying to fetch the mail queue 'spool' with the upgraded James memory image to branch-master.
+It seems there is potentially a bug where we loose the mail queues on memory version lately.
+Please fix it on James before re-enabling this test.
```
Reference the link to the james issue please

LanKhuat commented on src/main/scala-2.12/org/apache/james/gatling/jmap/rfc8621/JmapMailbox.scala:
```diff
@@ -1,5 +1,28 @@
+        .headers(JmapHttp.HEADERS_JSON)
+        .basicAuth("${username}", "${password}")
+      .body(StringBody(
+    s"""{
+       |  "using": [
+       |    "urn:ietf:params:jmap:core",
+       |    "urn:ietf:params:jmap:mail"
+       |  ],
+       |  "methodCalls": [[
+       |    "Mailbox/get",
+       |    {
+       |      "accountId": "$${accountId}",
+       |      "ids": null
+       |    },
+       |    "c1"]]
+       |}""".stripMargin)))
```
indent
